### ISSUE TITLE
static template instance

### ISSUE THREAD

**Issue by jogibear9988** [vanillawc/wc-marquee#3]
why don't you create the template as a static instance, so I could be reused?

**Comment by evanplaice** [comment]
A web component's constructor only runs once. The static method is just a convenient pattern to define a the contents of a template.

**Comment by jogibear9988** [comment]
it runs once per instance. so a cached template could increase perf on multiple uses

**Comment by evanplaice** [comment]
Yes, once per instance and there is only ever one global instance of a web component; which is created when `customElements.define()` is called.

The `connectedCallback` method applies on a per-mount/per-instance (to the DOM) basis. Even if it didn't, templates are also a global singeton-like element that only gets created once but can be re-used in many places. Neither Web Components or Templates act like typical DOM elements. They don't have the same limitations that typical framework components do.

In short, a web component's constructor only ever gets called once. `document.createElement('template')` only ever gets called once. There is no need to cache anything; by design, the platform already takes care of that.

**Comment by jogibear9988** [comment]
@evanplaice 
No the constructor is run for every instance
see: https://jsbin.com/nusagop/1/edit?html,js,output

**Comment by jogibear9988** [comment]
It's not a huge performance penalty, but it is.
